city_scrapers/spiders/il_liquor_control.py

```diff
@@ -0,0 +1,161 @@
+        "Hearing-Schedule-for-Chicago-IL-and-Springfield-IL.aspx",
+        "https://www2.illinois.gov/ilcc/Divisions/Pages/Legal/Meeting-minutes.aspx"
+    ]  # start_urls[0] contains links future meetings. start_urls[1] contains past
+    # meeting minutes links
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+
+        Each tentative meeting has own page. start_urls[0] is only used to get links for the pages.
+        The links are sent to _next_meeting.
+
+        start_urls[1] is for obtaining datetime info for past meetings. Then checks if
+        meeting detail page is still valid. If valid scrape through _prev_meeting_w_page. If not
+        valid, scrape available info from _prev_meeting_wo_page.
+        """
+
+        if response.url == self.start_urls[0]:
+            for future_meeting_href in response.css('div.soi-link-item a::attr(href)').extract():
+                yield response.follow(future_meeting_href, callback=self._next_meeting)
+
+        if response.url == self.start_urls[1]:
+            for idx, past_meeting_date in enumerate(response.css("div.link-item ::text").extract()):
+                dt_object = datetime.datetime.strptime(past_meeting_date, "%B %d, %Y Minutes")
+                minute_href = response.css("div.link-item a::attr(href)").extract()[idx]
+                meeting_url = ("https://www2.illinois.gov/ilcc/Events/Pages/Board-"
+                               "Meeting-{0}-{1}-{2:%y}.aspx".format(
+                                dt_object.month, dt_object.day, dt_object))
+                if requests.get(meeting_url).status_code == 200:
+                    yield response.follow(meeting_url,
+                                          callback=self._prev_meeting_w_page,
+                                          cb_kwargs=dict(minute_href=minute_href))
+                else:
+                    yield from self._prev_meeting_wo_page(response, dt_object, minute_href)
+
+    def _next_meeting(self, response):
+        dt_str = ' '.join([
+                x.strip() for x in response.css('div.soi-event-data::text').extract()
+            ])  # Extract string that contains datetime info.
+        meeting = Meeting(
+                title=self._parse_title(response),
+                description='',
+                classification=BOARD,
+                start=self._parse_start(dt_str),
+                end=self._parse_end(dt_str),
+                all_day=False,
+                time_notes='',
+                location=self._parse_location(response),
+                links=self._parse_link(response, self._parse_start(dt_str), None),
+                source=response.url,
+            )
+
+        meeting["status"] = self._get_status(meeting)
+        meeting["id"] = self._get_id(meeting)
+
+        yield meeting
+
+    def _prev_meeting_w_page(self, response, minute_href):
```

Comment:
This is a smart setup for passing arguments to the callback! One thing that might make it more obvious that you're using `kwargs` here is to use the Python convention for keyword arguments like this:

```
def _prev_meeting_w_page(self, response, **kwargs):
    minutes_href = kwargs["minutes_href"]
```

You don't have to assign the variable that way, but it's just to demonstrate that you'll have any keyword arguments available as keys in the `kwargs` dictionary